DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -500,6 +500,46 @@ struct printDataOrigin {
 };
 using DataOrigin = Descriptor<gSizeDataOriginString, printDataOrigin>;
 
+
+//__________________________________________________________________________________________________
+/// @struct DataIdentifier
+/// @brief Helper struct to encode origin and description of data.
+///
+/// The DataHeader stores origin and description of data in adedicted fields,
+/// DataIdentifier structure is used for assignment and comparison
+///
+/// @ingroup aliceo2_dataformats_dataheader
+struct DataIdentifier
+{
+  //a full data identifier combining origin and description
+  DataDescription dataDescription;
+  DataOrigin dataOrigin;
+  DataIdentifier();
+  DataIdentifier(const DataIdentifier&) = default;
+  DataIdentifier(const DataDescription &pDataDesc, const DataOrigin &pDataOrigin)
+    : dataDescription{pDataDesc}, dataOrigin{pDataOrigin}
+  {
+  }
+  template<std::size_t N, std::size_t M>
+  DataIdentifier(const char (&desc)[N], const char (&origin)[M])
+    : dataDescription(desc), dataOrigin(origin)
+  {
+  }
+
+  bool operator<(const DataIdentifier& other) const
+  {
+    if (dataDescription < other.dataDescription)
+      return true;
+    if (dataOrigin < other.dataOrigin)
+      return true;
+    return false;
+  }
```

Comment:
this would be useful for sorting, but how do you define (and need) ordering for headers?